ports/rp2/machine_uart.c

```diff
@@ -284,10 +423,10 @@ STATIC mp_uint_t machine_uart_ioctl(mp_obj_t self_in, mp_uint_t request, mp_uint
     if (request == MP_STREAM_POLL) {
         uintptr_t flags = arg;
         ret = 0;
-        if ((flags & MP_STREAM_POLL_RD) && uart_is_readable(self->uart)) {
+        if ((flags & MP_STREAM_POLL_RD) && (ringbuf_avail(&self->read_buffer) > 0)) {
```

Comment:
minor point: parenthesis are not needed around this `> 0` condition

Reply:
Removed